src/v/cluster/rm_stm.cc

```diff
@@ -1138,6 +1138,16 @@ rm_stm::get_expiration_info(model::producer_identity pid) const {
     return it->second;
 }
 
+std::optional<int32_t>
```

Comment:
> cluster: Add seq_numer to tx info in rm_stm

in your commit message, please describe why you are adding this information. for instance, if it is going to be used by a feature later in the PR then you could mention that feature.